### PR TITLE
Try: Add a subtle animation to the is-active indicator for sidebar tabs

### PR DESCRIPTION
**Pull request by kjellr**
This PR adds an animation to the `is-selected` indicator for sidebar tabs. In the animation, the bottom border subtly animates up from the bottom of the toolbar:

toolbar-tab

^ The GIF is slowed down so 3x so the frame rate can show the effect — it's actually much faster in practice.

To achieve a more performant animation, the `border` has been adjusted to use `box-shadow` instead. As per [this article](https://css-tricks.com/animating-border/), that helps prevent changing the height of the element during animation, requiring a bunch of unnecessary layout rewriting. 
 
_Related to #8029_

### CODE REVIEW

**Comment by jasmussen** [comment]
I love it! Really really slick work. There are a few small things we need to fix, but this is a delightful little detail.

First thing: in Safari the box shadow is actually offset very slightly by one pixel. Look very closely:

safari

Here's a higher fidelity screenshot:

screenshot 2019-02-20 at 10 25 17

This was on a retina screen. That might have something to do with it. 

In Chrome, the focus rectangle lingers, and also doesn't include the actual border. We'll probably want to rethink that a bit — one aspect might be to make the focus rectangle consistent with other recent focus rectangles:

screenshot 2019-02-20 at 10 27 28

screenshot 2019-02-20 at 10 27 21

The other bit is simply making sure that the box shadow is actually present on :focus, and not just on active.

I consider Google properties to be more or less the gold standard for web animation these days, and there's a lovely little detail on Google Translate where the tab indicator animates from 0px to full wide:

google translate

We don't have to do that, I don't even know how we would — but I like the _principle_ of this. The idea is that every single element comes from somewhere. This comes from nothing and scales into view. A fade is definitely similar in principle to this, so on that front this is cool.

**Comment by kjellr** [comment]
> This was on a retina screen. That might have something to do with it.

Ah, good catch. I built this on my non-retina monitor yesterday, but now that I'm back on my laptop I see that too. 

> In Chrome, the focus rectangle lingers, and also doesn't include the actual border.

Cool, I can fix this. 

> I consider Google properties to be more or less the gold standard for web animation these days, and there's a lovely little detail on Google Translate where the tab indicator animates from 0px to full wide:

Yeah! I looked at this, as well as this Google one: 

https://storage.googleapis.com/spec-host-backup/mio-design%2Fassets%2F1ZfWOjAaOgtMKDyJR6cMgWknWe-iemQ2S%2Ftabs-tap.mp4

And I actually was inspired to try this "up-from-the-bottom" approach after looking at the tab animation for [the Material handbook](https://material.io) itself: 

design-develop

From a quick look at some of Google's code, it seemed like in some cases, they're including an extra (empty) `span` in their tabs, which they can then animate around. That seemed a little weird to me, and this "up-from-the-bottom" approach seemed like a good mix of simplicity in motion and code.

**Comment by kjellr** [comment]
@jasmussen should be all set after those last two commits: 

tabs-fixed

- I removed that 1px of empty space. 
- I changed the focus outline to a dotted line.
- I ensured that the blue indicator will remain when an item is focused.

**Comment by jasmussen** [comment]
This is such a profound improvement. The focus style is now way more consistent than it ever was, and the animation is a complete delight! Thank you Kjell.

There's one final thing we need to fix, and it's for Windows High Contrast mode. Here's before:

master

Here's after:

high contrast

The new focus style is fine, but we need for the active tab indicator to show up in high contrast mode.

What happens is that high contrast mode extracts and removes any instance of `box-shadow`, and shows only `outline` and `border`. The trick to getting this to work well in both, is to use a _transparent_ border. You don't see this transparent border in normal contrast mode, but in high contrast mode, all transparent borders become fully opaque.

In other words, if you can find a way to work in a 3px solid transparent bottom border to indicate the active tab, then it will show up as intended in Windows High Contrast mode. After that, let's ship this thing!

**Comment by kjellr** [comment]
> What happens is that high contrast mode extracts and removes any instance of `box-shadow`, and shows only `outline` and `border`. The trick to getting this to work well in both, is to use a _transparent_ border. You don't see this transparent border in normal contrast mode, but in high contrast mode, all transparent borders become fully opaque.

Whoa, thanks for that tip! That's a weird one. 

This was a little tough to solve for, since the up-from-the-bottom animation requires use of an _inset_ shadow (otherwise, it'd animate from top-to-bottom instead). If I add a `3px` border to the bottom of the active tab, the inset shadow would have to start 3px up from the bottom. And that's no good: 

screen shot 2019-02-21 at 10 04 11 am

I was able to get this working by adding an absolutely positioned `:before` pseudo class with a transparent border. This showed up nicely in High Contrast mode, but didn't interfere with any functionality. 

_Chrome on Mac:_ 

focus-2

_IE11 with Windows High Contrast Mode:_ 

focus

This seems to solve the issue, and I don't see any real issues with this stray pseudo element. Let me know what you think!

**Review by jasmussen — APPROVED** [review]
Killer work, dude. I'm a big fan of your doings.
